client/my-sites/plugins/plugins-discovery-page/index.jsx

```diff
@@ -89,6 +112,12 @@ const PluginsDiscoveryPage = ( props ) => {
 	} );
 
 	const isLoggedIn = useSelector( isUserLoggedIn );
+	const siteId = useSelector( getSelectedSiteId );
+	const sitePartnerBundle = useSelector( ( state ) =>
+		getSiteOption( state, siteId, 'site_partner_bundle' )
+	);
+	const isWPBeginnerSpecial =
+		getQueryArgs()?.ref === 'wpbeginner-special-lp' || sitePartnerBundle === 'wpbeginner-special';
```

Comment:
`sitePartnerBundle === 'wpbeginner-special';` - this needs:
* the site to be created by going through the `/setup/new-hosted-site?partnerBundle=wpbeginner-special`
* this [Jetpack PR](https://github.com/Automattic/jetpack/pull/39905) needs to be merged